base/task.jl

```diff
@@ -4,6 +4,49 @@
 
 show(io::IO, t::Task) = print(io, "Task ($(t.state)) @0x$(hex(convert(UInt, pointer_from_objref(t)), WORD_SIZE>>2))")
 
+type CollectedException <: Exception
```

Comment:
`@async` is used for local I/O bound tasks too. I had a `RemoteException` with (pid, ex, bt) and a locally raised `CollectedException` with (ex, bt). Combined both of them into `CollectedException`.